### ISSUE TITLE
Impossible to create AlloyDB instance with PSC enabled

### ISSUE THREAD

**Issue by JPFrancoia** [hashicorp/terraform-provider-google#18280]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to a user, that user is claiming responsibility for the issue.
* Customers working with a Google Technical Account Manager or Customer Engineer can ask them to [reach out internally](https://github.com/hashicorp/terraform-provider-google/wiki/Customer-Contact#raising-gcp-internal-issues-with-the-provider-development-team) to expedite investigation and resolution of this issue.

### Terraform Version & Provider Version(s)

Terraform v1.5.7 on darwin_arm64 (Mac OS, M1 machine)
+ provider registry.terraform.io/hashicorp/google 9e7dede23 (It's the main branch pulled this morning around 10 am UK time)
on darwin_arm64 (Mac OS, M1 machine)

### Affected Resource(s)

AlloyDB instance, possibly AlloyDB cluster

### Terraform Configuration

```tf
resource "google_alloydb_cluster" "test_cluster" {
  cluster_id   = "test-cluster"
  location     = var.region
  database_version = "POSTGRES_15"

  initial_user {
    user     = "alloy_db_user"
    password = "password"
  }

  ## Works if provided instead of psc_config block
  #network_config {
    #network = google_service_networking_connection.private_vpc_connection.network
  #}

  psc_config {
    psc_enabled = true
  }
}


resource "google_alloydb_instance" "test_instance" {
  cluster       = google_alloydb_cluster.test_cluster.name
  instance_id   = "alloydb-instance"
  instance_type = "PRIMARY"

  database_flags = {
    "alloydb.logical_decoding" = "on" # Enable logical decoding for Datastream
  }

  machine_config {
    cpu_count = 2
  }

  #Same error with or without this block
  psc_instance_config {
      allowed_consumer_projects = ["979541276525"]
  }
}
```

### Debug Output

https://gist.github.com/JPFrancoia/6ab6ee3890e62eb6a3d1b816ece714b0

### Expected Behavior

I expected an AlloyDB instance to be created in my AlloyDB cluster

### Actual Behavior

The AlloyDB instance wasn't created, the terraform apply failed

### Steps to reproduce

Build the main branch of this repo (commit 9e7dede23)

```
git clone https://github.com/hashicorp/terraform-provider-google.git
cd terraform-provider-google
go build -o terraform-provider-google
mkdir -p ~/.terraform.d/plugins/terraform.local/local/google/5.32.0/darwin_arm64
mv terraform-provider-google ~/.terraform.d/plugins/terraform.local/local/google/5.32.0/darwin_arm64
```

Then run `terraform apply` on the config I pasted earlier, and you should get the output pasted in the gist.

### Important Factoids

For reasons that are out of scope for this issue, I need to create a Private Service Connect (PSC) endpoint for my AlloyDB database. The idea is to then use this endpoint to connect from a Cloud Run service. I'm aware that there are other solutions to connect CR and Alloy, but due to the network architecture of my company, I don't really have a choice here.

PSC isn't supported for Alloy on the most recent version of the provider (5.31.1 at the time of this issue). Luckily, it landed in main just yesterday: https://github.com/hashicorp/terraform-provider-google/pull/18263/files. Thanks to [@vmiglani ](https://github.com/vmiglani) for that.

Their PR on the magic modules repo is here: https://github.com/GoogleCloudPlatform/magic-modules/pull/10783.

Unfortunately, I'm unable to get it to work.

Does anyone have an idea to unlock me? Really keen to support testing.

**Comment by ggtisc** [comment]
This looks like a new feature that isn't published yet on `terraform registry` or `Google Cloud documentation` and it isn't possible to reproduce.

I'm forwarding this for its validation

**Comment by JPFrancoia** [comment]
Yeah I think there is a bug in the Alloy DB API on Google's side, I don't think it's a terraform issue. If they fix the API this bug should be fixed too. This issue could still be useful though, if a release is made from the main branch now, anyone trying to deploy a cluster with PSC enabled will encounter this issue (until Google fixes the API).

**Comment by vmiglani** [comment]
This is an issue with the VPC-SC perimeter of this project due to their organization level VPC-SC setting. We are following up with the VPC-SC team to understand how this project can be unblocked from creating PSC-enabled AlloyDB clusters.

We have verified that AlloyDB cluster creation works correctly for projects that don't have VPC-SC restrictions. AlloyDB will be adding support for VPC-SC compliance in the near future.
